File: frontend/src/pages/Favorite.js
import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import majorsData from '../majors.json';
import favoriteIcon from '../assets/favorite.png';
import './Favorite.css';

const BASE_KEY = 'favorites';

// ✅ Use userId or email from localStorage/session
function getUserKey() {
  const user = JSON.parse(localStorage.getItem('currentUser')); 
  // Example: { id: 1, email: "test@example.com" }
  return user ? `${BASE_KEY}_${user.id}` : BASE_KEY;
}

// Load favorites for this user
function loadFavorites() {
  try {
    return JSON.parse(localStorage.getItem(getUserKey())) || [];
  } catch {
    return [];
  }
}

// Save favorites for this user
function saveFavorites(list) {
  localStorage.setItem(getUserKey(), JSON.stringify(list));
}

function Favorite() {
  const [favorites, setFavorites] = useState([]);
  const navigate = useNavigate();

  // Load favorite majors from localStorage
  const updateFavorites = () => {
    const favIds = loadFavorites();
    const favMajors = majorsData.filter(m => favIds.includes(m.pk));
    setFavorites(favMajors);
  };

  useEffect(() => {
    updateFavorites();
  }, []);

  const handleToggle = (pk) => {
    let favs = loadFavorites();
    if (favs.includes(pk)) {
      favs = favs.filter(id => id !== pk);
    } else {
      favs.push(pk);
    }
    saveFavorites(favs);
    updateFavorites();
  };

  const handleCardClick = (pk) => {
    navigate(`/MajorsView/${pk}`);
  };

  return (
    <div className="favorites-container">
      <h1>Your Favorite Majors</h1>

      {favorites.length === 0 ? (
        <p>No favorites yet.</p>
      ) : (
        <div className="majors-grid">
          {favorites.map(major => (
            <div key={major.pk} className="major-card">
              <div
                className="card-content"
                onClick={() => handleCardClick(major.pk)}
              >
                <h2>{major.fields.name}</h2>
                <p>{major.fields.about}</p>
              </div>
              <button
                onClick={() => handleToggle(major.pk)}
                className="fav-btn active"
              >
                <img src={favoriteIcon} alt="Favorite" className="fav-icon" />
                Remove Favorite
              </button>
            </div>
          ))}
        </div>
      )}
    </div>
  );
}

export default Favorite;

File: backend/routes/favorite.py
# import json
# import os

# BASE_KEY = 'favorites'

# def get_user_key():
#     user = json.loads(os.getenv('currentUser', '{}'))
#     return f"{BASE_KEY}_{user['id']}" if user else BASE_KEY

# def load_favorites():
#     try:
#         return json.loads(os.getenv(get_user_key(), '[]'))
#     except:
#         return []

# def save_favorites(lst):
#     os.environ[get_user_key()] = json.dumps(lst)

# class Favorite:
#     def __init__(self):
#         self.favorites = []
#         self.update_favorites()

#     def update_favorites(self):
#         fav_ids = load_favorites()
#         fav_majors = [m for m in majors_data if m['pk'] in fav_ids]
#         self.favorites = fav_majors

#     def handle_toggle(self, pk):
#         favs = load_favorites()
#         if pk in favs:
#             favs.remove(pk)
#         else:
#             favs.append(pk)
#         save_favorites(favs)
#         self.update_favorites()

#     def handle_card_click(self, pk):
#         navigate(f"/MajorsView/{pk}")

#     def render(self):
#         print("Your Favorite Majors")
#         if not self.favorites:
#             print("No favorites yet.")
#         else:
#             for major in self.favorites:
#                 print(f"Major: {major['fields']['name']}")
#                 print(f"About: {major['fields']['about']}")
#                 print("Remove Favorite")